fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFab.kt

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.FloatingActionButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Floating action button for tabs tray.
+ *
+ * @param tabsTrayStore [TabsTrayStore] used to listen for changes to [TabsTrayState].
+ */
+@Composable
+fun TabsTrayFab(
+    tabsTrayStore: TabsTrayStore,
+) {
+    val trayState = tabsTrayStore
+        .observeAsComposableState { state ->
+            TrayState(
+                state.selectedPage,
+                state.syncing,
+                state.mode == TabsTrayState.Mode.Normal,
+            )
+        }.value ?: TrayState()
+    val icon = painterResource(
+        id = if (trayState.currentPage == Page.SyncedTabs) {
+            R.drawable.ic_fab_sync
+        } else {
+            R.drawable.ic_new
+        },
+    )
+    val contentDescription = stringResource(
+        id = when (trayState.currentPage) {
+            Page.SyncedTabs -> R.string.tab_drawer_fab_sync
+            Page.PrivateTabs -> R.string.add_private_tab
+            else -> R.string.add_tab
+        },
+    )
+    val label = when (trayState.currentPage) {
+        Page.SyncedTabs -> if (trayState.isSyncing) {
+            stringResource(id = R.string.sync_syncing_in_progress)
+        } else {
+            stringResource(id = R.string.resync_button_content_description)
+        }
+        Page.PrivateTabs -> stringResource(id = R.string.tab_drawer_fab_content)
+        else -> null
+    }?.uppercase()
```

Comment:
alternatively, this could be a singular `when` based on the current page, and have each val initialized inside. something like:

```suggestion
    val icon: PainterResource
    val contentDescription: String
    val label: String
    
    when (trayState.currentPage) {
            Page.SyncedTabs -> {
                icon = ...
                contentDescription = ...
                label = ...
            }
            Page.PrivateTabs -> {
                icon = ...
                contentDescription = ...
                label = ...
            }
            Page.NormalTabs -> {
                 icon = ...
                contentDescription = ...
                label = ...
             }
    }
```